### DISCUSSION TITLE
Error: Failed to launch the browser process! /tmp/chromium: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory

### DISCUSSION THREAD

**Discussion by jsbimra** [vercel/community#4273] [Help]
### Page to Investigate

https://bizstore.doableyo.com/

### Steps to Reproduce

Login would require to reproduce the error. Earlier it was working fine after vercel node.js environment upgraded node to 16 or 18 it stopped working, any possible solution or guide to look?

but here is the below full log from server:
Error: Failed to launch the browser process!
/tmp/chromium: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory

TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md

    at onClose (/var/task/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:203:20)
    at Interface.<anonymous> (/var/task/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:193:68)
    at Interface.emit (node:events:525:35)
    at Interface.close (node:readline:590:8)
    at Socket.onend (node:readline:280:10)
    at Socket.emit (node:events:525:35)
    at endReadableNT (node:internal/streams/readable:1358:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)

**Comment by amyegan** [comment]
Please double check that the Node and Puppeteer versions are compatible. Package version conflicts are a common cause of issues after a Node.js major version upgrade. 🙏

**Reply by jsbimra** [reply]
is it something related too : 
When you install Puppeteer, it automatically downloads a recent version of [Chrome for Testing](https://goo.gle/chrome-for-testing) (~170MB macOS, ~282MB Linux, ~280MB Windows) that is [guaranteed to work](https://pptr.dev/faq#q-why-doesnt-puppeteer-vxxx-work-with-chromium-vyyy) with Puppeteer. The browser is downloaded to the $HOME/.cache/puppeteer folder by default (starting with Puppeteer v19.0.0).

If you deploy a project using Puppeteer to a hosting provider, such as Render or Heroku, you might need to reconfigure the location of the cache to be within your project folder (see an example below) because not all hosting providers include $HOME/.cache into the project's deployment.
https://pptr.dev/#installation ? Trying to figure out the version compatibility though, thanks for pointing


open issue on issuetracker google https://issuetracker.google.com/u/3/issues/266279679 looks no solution still

**Comment by jsbimra** [comment]
Updated to puppeteer-chromium-resolver,

same code working fine on localhost, on vercel getting this:
…oogleapis.com ... [31m[PCR] Error: ENOENT: no such file or directory,
mkdir '/home/sbx_user1051' [39m [PCR] Downloading from:
https://npmmirror.com/mirrors ... [31m[PCR] Error: ENOENT: no such file or
directory, mkdir '/home/sbx_user1051' [39m [PCR] Retry Chromium downloading
... [PCR] Downloading from: https://storage.googleapis.com ... [31m[PCR]
Error: ENOENT: no such file or directory, mkdir '/home/sbx_user1051' [39m
[PCR] Downloading from: https://npmmirror.com/mirrors ... [31m[PCR] Error:
ENOENT: no such file or directory, mkdir '/home/sbx_user1051' [39m [PCR]
Retry Chromium downloading ... [PCR] Downloading from:
https://storage.googleapis.com ... [31m[PCR] Error: ENOENT: no such file or
directory, mkdir '/home/sbx_user1051' [39m [PCR] Downloading from:
https://npmmirror.com/mirrors ... [31m[PCR] Error: ENOENT: no such file or
directory, mkdir '/home/sbx_user1051' [39m [PCR] Retry Chromium downloading
... [PCR] Downloading from: https://storage.googleapis.com ... [31m[PCR]
Error: ENOENT: no such file or directory, mkdir '/home/sbx_user1051' [39m
[PCR] Downloading from: https://npmmirror.com/mirrors ... [31m[PCR] Error:
ENOENT: no such file or directory, mkdir '/home/sbx_user1051' [39m
[31m[PCR] ERROR: Failed to download Chromium after retry 3 times. [39m
[PCR] Chromium launchable: [31mfalse [39m [31m[PCR] Failed to save stats:
/home/sbx_user1051/.pcr-stats.json [39m Error: An `executablePath` or
`channel` must be specified for `puppeteer-core` at assert
(/var/task/node_modules/puppeteer-core/lib/cjs/puppeteer/util/assert.js:28:15)
at ChromeLauncher.computeLaunchArguments
(/var/task/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:96:36)
at ChromeLauncher.launch
(/var/task/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:79:28)
at _default (/var/task/api/generate-invoice.js:50:25) at Server.<anonymous>
(/var/task/___vc/__launcher/helpers-23SL6TQX.js:1:7888) at
Server.<anonymous> (/var/task/___vc/__launcher/__launcher.js:14:9173)
TypeError: Cannot read properties of undefined (reading 'newPage') at
_default (/var/task/api/generate-invoice.js:60:36) at Server.<anonymous>
(/var/task/___vc/__launcher/helpers-23SL6TQX.js:1:7888) at
Server.<anonymous> (/var/task/___vc/__launcher/__launcher.js:14:9173)

any help?

**Comment by vikiival** [comment]
[So if you are looking for a working version in 2024](https://github.com/vikiival/vercelgl)

TL;DR

```json
  "dependencies": {
    "@sparticuz/chromium": "^119.0.2",
    "puppeteer-core": "^21.5.1"
  },
  "engines": {
    "node": "18"
  }
```

**Reply by Vanals** [reply]
Is there a working version for Node20?

**Comment by jsbimra** [comment]
![image](https://github.com/vercel/community/assets/1294763/a83463c1-9130-4153-8ea8-4fef28005df5)

`
"description": "fe-gen-template",
  "devDependencies": {
    "@sparticuz/chromium": "^123.0.1",
    "geolib": "^3.3.3",
    "jest": "^29.5.0",
    "puppeteer-core": "^22.9.0"
  },

`
Implementation code here
https://gist.github.com/jsbimra/f5892da8047f05399c3729fe7721fe49  
  
  still throw error any quick turn around?
